Packs/AWS-SQS/Integrations/AWS-SQS/AWS-SQS.py

```diff
@@ -176,15 +180,20 @@ def fetch_incidents(aws_client, aws_queue_url, max_fetch, parse_body_as_json):
                     break
 
             for message in messages["Messages"]:
-                receipt_handles.append(message['ReceiptHandle'])
+                receipt_handles.add(message['ReceiptHandle'])
+                # Checking to avoid duplicates
+                if last_receipt_handles and message['ReceiptHandle'] in last_receipt_handles:
+                    continue
                 incidents.append(parse_incident_from_finding(message, parse_body_as_json))
                 incidents_created += 1
                 if incidents_created == max_fetch:
                     break
 
-            demisto.incidents(incidents)
-            for receipt_handle in receipt_handles:
-                client.delete_message(QueueUrl=aws_queue_url, ReceiptHandle=receipt_handle)
+        demisto.setLastRun({"lastReceiptHandles": receipt_handles})
+        demisto.debug('last_receipt_handles after fetch occurred" -> {} {}'.format(len(receipt_handles), receipt_handles))
+        demisto.incidents(incidents)
```

Comment:
Please replace the order between `setLastRun` and `incidents` - if `demisto.incidents()` fails for some reason, it will try again later (thus not skipping any incidents).